src/gui/settings.cpp

```diff
@@ -3158,18 +3435,27 @@ void FurnaceGUI::drawSettings() {
     if (ImGui::Button("OK##SettingsOK")) {
       settingsOpen=false;
       willCommit=true;
+      settingsChanged=false;
     }
     ImGui::SameLine();
     if (ImGui::Button("Cancel##SettingsCancel")) {
       settingsOpen=false;
       audioEngineChanged=false;
       syncSettings();
+      settingsChanged=false;
     }
     ImGui::SameLine();
+    bool applyDisabled=false;
+    if (!settingsChanged) {
+      ImGui::BeginDisabled();
```

Comment:
You don't have to define an extra variable. Just pass (!settingsChanged) to BeginDisabled.